src/tileray.cpp

```diff
@@ -3,13 +3,14 @@
 #include <cmath>
 #include <cstdlib>
 
+#include "line.h"
 #include "math_defines.h"
 
 static const int sx[4] = { 1, -1, -1, 1 };
 static const int sy[4] = { 1, 1, -1, -1 };
 
-tileray::tileray(): deltax( 0 ), deltay( 0 ),  leftover( 0 ), direction( 0 ),
-    last_dx( 0 ), last_dy( 0 ), steps( 0 ), infinite( false )
+tileray::tileray(): delta( point_zero ), leftover( 0 ), direction( 0 ),
+    last_d( point_zero ), steps( 0 ), infinite( false )
```

Comment:
Probably no need to initialize `delta` and `last_d` as points are initialized as `point_zero` by default?

Reply:
Indeed.  Fixed.

(Wrote this code before I wrote that check).